server/src/main/java/org/candlepin/pinsetter/core/PinsetterJobListener.java

```diff
@@ -129,6 +130,10 @@ private void updateJob(JobExecutionContext ctx, JobExecutionException exc) {
                 log.error("Job [" + status.getId() + "] failed." , exc);
                 status.setState(JobState.FAILED);
                 status.setResult(exc.getMessage());
+
+                if (exc.getCause() instanceof CandlepinException) {
+                    status.setResultData(((CandlepinException) exc.getCause()).message());
```

Comment:
I'm not sure I fully understand, but I'll respond based on what I *think* you mean. :)

CandlepinException.message() returns an ExceptionMessage which is made up of a bunch of extra data (based on implementation).  status.result is intended to be the original message string, but the result data will contain the ExceptionMessage object that will be serialized to json as part of the response. For import, this allows returning the conflict keys that can be overridden.